src/devices/video/nereid.cpp

```diff
@@ -0,0 +1,101 @@
+// license:BSD-3-Clause
+// copyright-holders:Sven Schnelle
+
+#include "emu.h"
+#include "nereid.h"
+
+//#define VERBOSE 1
+#include "logmacro.h"
+
+DEFINE_DEVICE_TYPE(NEREID, nereid_device, "nereid", "HP Nereid ASIC")
+
+nereid_device::nereid_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock)
+	: device_t(mconfig, type, tag, owner, clock),
+	m_red(0),
+	m_green(0),
+	m_blue(0),
+	m_index(0),
+	m_plane_mask(0)
+{
+}
+
+nereid_device::nereid_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: nereid_device(mconfig, NEREID, tag, owner, clock)
+{
+}
+
+void nereid_device::device_start()
+{
+}
+
+void nereid_device::device_reset()
+{
+	memset(&m_palette, 0, sizeof(m_palette));
+}
+
+READ16_MEMBER(nereid_device::ctrl_r)
+{
+	uint8_t ret = 0;
+
+	LOG("NEREID: %02X\n", offset);
+
+	switch(offset) {
+	case NEREID_RED_DATA:
+		ret = m_red;
+		break;
+	case NEREID_GREEN_DATA:
+		ret = m_red;
+		break;
+	case NEREID_BLUE_DATA:
+		ret = m_red;
+		break;
+	case NEREID_INDEX:
+		ret = m_index;
+		break;
+	case NEREID_STROBE:
+		ret = 0;
+		break;
+	case NEREID_PLANE_MASK:
+		ret = m_plane_mask;
+		break;
+	default:
+		space.unmap();
```

Comment:
You want to do something with the result here?  In general I dislike this pattern.  You're initialising `ret` even though you always plan to assign to it.  If you *didn't* initialise it you'd get a compiler warning for this.  Also, does the `ret` variable really gain anything over a `return` in each case?

Reply:
Propably a matter of personal preference, but i changed it, thanks.